azure-configurationservice/tests/test_azure_configuration_client.py

```diff
@@ -0,0 +1,601 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import datetime
+from copy import copy
+import pytest
+
+
+from msrestazure.azure_exceptions import CloudError
+from azure.configuration import AzureConfigurationClient
+from azure.configuration import ConfigurationSetting
+from devtools_testutils import AzureMgmtTestCase
+
+import conftest
+
+
+class AzConfigTestData:
+    def __init__(self):
+        self.key_uuid = None
+        self.label1 = None
+        self.label2 = None  # contains reserved chars *,\
+        self.label_uuid = None
+        self.label1_data = []
+        self.label2_data = []
+        self.no_label_data = []
+
+
+class AzConfigurationClientTest(AzureMgmtTestCase):
+    def setUp(self):
+        self.working_folder = os.path.dirname(__file__)
+        super(AzConfigurationClientTest, self).setUp()
+
+        connection_str = os.environ["AZ_CONFIG_CONNECTION"]
+        self.app_config_client = AzureConfigurationClient(connection_str)
+        self.test_data = conftest.setup_data()
+
+    def tearDown(self):
+        conftest.teardown_data(self.test_data)
+
+    # method: add_configuration_setting
+    def test_add_configuration_setting(self):
+
+        kv = ConfigurationSetting()
+        kv.key = "unit_test_key_add_" + self.test_data.key_uuid
+        kv.label = self.test_data.label1
+        kv.value = "test value"
+        kv.content_type = "test content type"
+        kv.tags = {"tag1": "tag1", "tag2": "tag2"}
+        kv.etag = "not needed"  # this etag will not be processed.
+        created_kv = self.app_config_client.add_configuration_setting(kv)
+        self.app_config_client.delete_configuration_setting(kv.key, kv.label)
```

Comment:
what happens if the delete operation fails?

Reply:
Then this test method and some other test methods will fail. This is against the theory of unit test, meaning all unit test methods should be isolated. This unit test is actually kind of "integration" test, and this is not very likely to happen. So I think this is acceptable

Reply:
What I mean is that if the delete method fails, then the assert will never happen, which is what the test should be testing. 
Better to assert and make sure the test is right, and then you can do all the cleanup that you need

Reply:
Got you. Yes it's better to do cleanup after assert.